make/integration-tests.mk

```diff
@@ -45,7 +45,7 @@ test-gatewayapi-env-integration: clean-cov generate fmt vet ginkgo ## Requires k
 test-istio-env-integration: clean-cov generate fmt vet ginkgo ## Requires kubernetes cluster with GatewayAPI and Istio installed.
 	mkdir -p $(PROJECT_PATH)/coverage/istio-integration
 #	Check `ginkgo help run` for command line options. For example to filtering tests.
-	$(GINKGO) \
+	GATEWAYAPI_PROVIDER=istio $(GINKGO) \
```

Comment:
just wondering if `ldflags` would be a better alternative. Ginkgo supports passing same flags as `go test`.

Something like 

```
ginkgo -ldflags "-X 'github.com/kuadrant/kuadrant-operator/tests.GatewayClassName=istio'" 
```

Reply:
This is an interesting suggestion, I'm not sure which one I prefer. I guess in the code instead of retrieving / checking for the presence of the env var we would check that the `GatewayClassName` is set. The only thing I can think is that the env var would be easier if you're ever running the ginkgo command directly but I assume we would always be using the Makefile

Reply:
no strong opinion here. It is about having the value set at compile time vs runtime.